velox/connectors/hive/HiveDataSource.cpp

```diff
@@ -681,6 +681,7 @@ void HiveDataSource::addSplit(std::shared_ptr<ConnectorSplit> split) {
   configureRowReaderOptions(
       rowReaderOpts_,
       ROW(std::vector<std::string>(fileType->names()), std::move(columnTypes)));
+  rowReader_.reset();
```

Comment:
This is so important that we should put a comment above it so people don't remove it by mistake

Reply:
+1

Reply:
```
// NOTE: we first reset the finished 'rowReader_' of previous split before set up the next one to reduce the peak memory usage.
```